File: lib/RallyQCUtils/excel_writer.rb
require "win32ole"

module RallyQCUtils

  class ExcelWriter
    
      attr_reader :file_name, :current_row
    
      def initialize(file_name)
        begin
          @file_name = "#{file_name}.xlsx"
          @xl = WIN32OLE.new('Excel.Application')
          @book = nil
          @current_row = {"Data" => 1, "Rally"=>1, "HPQC" => 1}
          create_file
        rescue Exception => ex
          excel_rescue(ex.message)
        end
      end
    
      #def write_header_row(row_data)
      #  write_row(1, row_data)
      #end
    
      def add_row(sheetname, row_data)
        @current_row[sheetname] += 1
        write_row(sheetname,@current_row[sheetname], row_data)
      end
    
      def write_row(sheetname, row_num, row_data)
        col_count = 0
        row_data.each do |cellval|
          col_count += 1
          case sheetname
            when "Data"
               write_cell(@sheet1,{"row" => row_num, "column" => col_count}, cellval)
            when "Rally"
               write_cell(@sheet2,{"row" => row_num, "column" => col_count}, cellval)
            when "HPQC"
              write_cell(@sheet3,{"row" => row_num, "column" => col_count}, cellval)
          end
        end
      end
    
      def close_file
        @sheet1.Columns.Autofit
        @book.Save
        @book.Close
        @xl.Quit
        @book = nil
      end
    
def write(sheetname, data)
  case sheetname
    when "Rally"
      add_row("Rally",["workspace name", "workspace id", "project name", "project id"])
      data.each do |values|
        workspace_name = values[:name]
        workspace_id = values[:id]
        begin
        values[:projects].each do |project|
          add_row("Rally",["#{workspace_name}","#{workspace_id}","#{project[:name]}","#{project[:id]}"])
        end
        rescue Exception => ex
          puts "exception was #{ex.message}"
        end
      end
    when "QC"
      header_data = ["domain","project"].concat(data[0].slice(2..data.size))
      add_row("HPQC",header_data)
      data.each do |values|
        add_row("HPQC",[values[0], values[1]])
      end
  end
end

#------------------------------------------------------------------------------------
    
      private
    
      def create_file
        @book = @xl.Workbooks.Add
        @sheet1 = @book.WorkSheets(1)
        @sheet2 = @book.WorkSheets(2)
        @sheet3 = @book.WorkSheets(3)
        @sheet1.Name = "Data"
        @sheet2.Name = "Rally"
        @sheet3.Name= "HPQC"
        
        current_dir = Dir.pwd.gsub("/","\\")
        puts "#{current_dir}\\#{file_name} - created."
        @book.SaveAs("#{current_dir}\\#{@file_name}")
      end
    
      def write_cell(sheet, location, value)
        begin
          sheet.Cells(location["row"],location["column"]).Value = value
        rescue Exception => ex
          excel_rescue(ex.message)
          puts ex.backtrace
        end
      end
    
      def excel_rescue(msg)
        puts "An error occurred working with Excel - the message was:  #{msg}"
      end
    

  end
end

File: lib/RallyQCUtils/excel_writer.py

import xlwt
import sys

class ExcelWriter(object):
    def __init__(self,file_name):
        self.file_name = file_name+".xlsx"
        self.current_row = {"Data" : 1, "Rally" : 1, "HPQC" : 1}
        self.create_file()
        
    def add_row(self,sheetname, row_data):
        self.current_row[sheetname] += 1
        self.write_row(sheetname,self.current_row[sheetname], row_data)
    
    def write_row(self, sheetname, row_num, row_data):
        col_count = 0
        for cellval in row_data:
            col_count += 1
            if sheetname == "Data":
               self.write_cell(self.sheet1,{"row" : row_num, "column" : col_count}, cellval)
            elif sheetname=="Rally":
               self.write_cell(self.sheet2,{"row" : row_num, "column" : col_count}, cellval)
            elif sheetname=="HPQC":
               self.write_cell(self.sheet3,{"row" : row_num, "column" : col_count}, cellval)
    
    def close_file(self):
        self.book.save(self.file_name)
    
    def write(self,sheetname, data):
        if sheetname == "Rally":
            self.add_row("Rally",["workspace name", "workspace id", "project name", "project id"])
            for values in data:
                workspace_name = values["name"]
                workspace_id = values["id"]
                for project in values["projects"]:
                    self.add_row("Rally",[workspace_name,workspace_id,project["name"],project["id"]])
        elif sheetname== "HPQC":
            for values in data:
                self.add_row("HPQC",values)
    
    def create_file(self):
        self.book = xlwt.Workbook()
        self.sheet1 = self.book.add_sheet("Data")
        self.sheet2 = self.book.add_sheet("Rally")
        self.sheet3 = self.book.add_sheet("HPQC")
            
    def write_cell(self,sheet, location, value):
        sheet.write(location["row"],location["column"], value)
def main():
    filename = sys.argv[1]
    rally_data = eval(sys.argv[2])
    hpqc_data= eval(sys.argv[3])
    excel = ExcelWriter(filename)
    excel.write("Rally",rally_data)
    excel.write("HPQC",hpqc_data)
    excel.close_file()
    
if __name__ == '__main__':
    main()